tests/unit/lib/charms/vault_k8s/v0/test_vault_managers.py

```diff
@@ -0,0 +1,142 @@
+        ],
+    )
+    @patch("charms.vault_k8s.v0.vault_managers.VaultClient")
+    @patch("charms.vault_k8s.v0.vault_managers.JujuFacade")
+    def test_when_get_vault_configuration_details_called_then_details_are_retrieved_correctly(
+        self,
+        juju_facade_mock,
+        vault_client_mock,
+        token,
+        token_valid,
+        expected_token,
+    ):
+        juju_facade_instance = juju_facade_mock.return_value
+        charm = MagicMock()
+        tls_manager = MagicMock(spec=VaultTLSManager)
+        tls_manager.get_tls_file_path_in_workload.return_value = "/my/test/path"
+        vault_client_instance = vault_client_mock.return_value
+        vault_client_instance.token = token
+
+        def authenticate(auth_method):
+            if token and vault_client_instance.authenticate.call_count == 1:
+                return token_valid
+            vault_client_instance.token = "new token"
+            return True
+
+        vault_client_instance.authenticate.side_effect = authenticate
+        requires = MagicMock()
+        requires.get_details.return_value = AutounsealDetails(
+            "my_address",
+            "my_mount_path",
+            "my_key_name",
+            "my_role_id",
+            "my_secret_id",
+            "my_ca_certificate",
+        )
+        relation_id = 1
+        relation = MagicMock()
+        relation.id = relation_id
+        if token:
+            juju_facade_instance.get_secret_content_values.return_value = (token,)
+        if not token:
+            juju_facade_instance.get_secret_content_values.side_effect = SecretRemovedError()
+
+        autounseal = VaultAutounsealRequirerManager(charm, tls_manager, requires)
+        details = autounseal.get_vault_configuration_details()
+
+        assert details is not None
+        assert details.address == "my_address"
+        assert details.mount_path == "my_mount_path"
+        assert details.key_name == "my_key_name"
+        assert details.token == expected_token
+        assert details.ca_cert_path == "/my/test/path"
+
+
+class TestVaultAutounsealProviderManager:
+    def test_create_credentials(self):
+        charm = MagicMock()
+        provides = MagicMock()
+        relation_id = 1
+        relation = MagicMock()
+        relation.id = relation_id
+        vault_client = MagicMock(spec=VaultClient)
+        vault_client.create_or_update_approle.return_value = "role_id"
+        vault_client.generate_role_secret_id.return_value = "secret_id"
+
+        autounseal = VaultAutounsealProviderManager(charm, vault_client, provides, "ca_cert", 8200)
+
+        key_name, role_id, secret_id = autounseal.create_credentials(relation)
+
+        assert key_name == str(relation_id)
+        assert role_id == "role_id"
+        assert secret_id == "secret_id"
+        provides.set_autounseal_data.assert_called_once()
+
+    @patch("charms.vault_k8s.v0.vault_managers.JujuFacade")
+    def test_sync(self, juju_facade_mock):
```

Comment:
Let's follow the test naming convention we have `given_x_when_sync_then_z`

Reply:
Removed in https://github.com/canonical/vault-k8s-operator/pull/534/commits/b73344ca50ae70673a93f4552c470acdd2fd54f4